flight/PiOS/Common/pios_mpu.c

```diff
@@ -771,9 +772,9 @@ int32_t PIOS_MPU_SPI_Init(pios_mpu_dev_t *dev, uint32_t spi_id, uint32_t slave_n
 	mpu_dev->com_slave_addr = slave_num;
 	mpu_dev->cfg = cfg;
 
-	PIOS_MPU_SPI_Probe(&mpu_dev->mpu_type);
-
-	// TODO: detect mag?
+	if (PIOS_MPU_SPI_Probe(&mpu_dev->mpu_type)) {
+		return -1;
```

Comment:
Or alternately return the error from the call (it may return others, don't remember/phone is inconvenient to check).